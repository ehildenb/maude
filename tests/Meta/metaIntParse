#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/metaIntParse.maude -no-banner -no-advise \
  > metaIntParse.out 2>&1

diff $srcdir/metaIntParse.expected metaIntParse.out > /dev/null 2>&1
