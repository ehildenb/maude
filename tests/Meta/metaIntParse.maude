set show timing off .
set show advisories off .

***
*** Test parsing in the meta-interpreter.
***

load metaInterpreter

fmod FOO is
  inc RAT .
  sort Foo .
  op 0 : -> Foo .
endfm

mod PARSE-TEST is
  pr META-INTERPRETER .
  pr RAT .

  op me : -> Oid .
  op User : -> Cid .
  op term:_ : Term -> Attribute .
  op qids:_ : QidList  -> Attribute .

  vars X Y Z : Oid .
  var AS : AttributeSet .
  var QL : QidList .

  rl < X : User | AS > createdInterpreter(X, Y, Z) =>
     < X : User | AS > insertModule(Z, X, upModule('FOO, true)) .

  rl < X : User | AS, qids: QL > insertedModule(X, Y) =>
     < X : User | AS > parseQidList(Y, X, 'FOO, QL, anyType) .
endm


erew in PARSE-TEST : <> 
< me : User | qids: '1/2 > 
createInterpreter(interpreterManager, me, none) .

erew in PARSE-TEST : <> 
< me : User | qids: ('1/2 '+ 'X:Rat) > 
createInterpreter(interpreterManager, me, none) .

erew in PARSE-TEST : <> 
< me : User | qids: ('1/2 '+ 'bad) > 
createInterpreter(interpreterManager, me, none) .

erew in PARSE-TEST : <> 
< me : User | qids: ('1/2 '- '1) > 
createInterpreter(interpreterManager, me, none) .

erew in PARSE-TEST : <> 
< me : User | qids: ('1/2 '- '1 '- '1 '/ '2) > 
createInterpreter(interpreterManager, me, none) .

erew in PARSE-TEST : <> 
< me : User | qids: '0 > 
createInterpreter(interpreterManager, me, none) .
