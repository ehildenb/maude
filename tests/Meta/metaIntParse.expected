==========================================
erewrite in PARSE-TEST : <> < me : User | qids: '1/2 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 5
result Configuration: <> < me : User | none > parsedQidList(me, interpreter(0),
    {'_/_['s_['0.Zero],'s_^2['0.Zero]],'PosRat})
==========================================
erewrite in PARSE-TEST : <> < me : User | qids: ('1/2 '+ 'X:Rat) >
    createInterpreter(interpreterManager, me, none) .
rewrites: 5
result Configuration: <> < me : User | none > parsedQidList(me, interpreter(0),
    {'_+_['_/_['s_['0.Zero],'s_^2['0.Zero]],'X:Rat],'Rat})
==========================================
erewrite in PARSE-TEST : <> < me : User | qids: ('1/2 '+ 'bad) >
    createInterpreter(interpreterManager, me, none) .
rewrites: 5
result Configuration: <> < me : User | none > parsedQidList(me, interpreter(0),
    noParse(2))
==========================================
erewrite in PARSE-TEST : <> < me : User | qids: ('1/2 '- '1) >
    createInterpreter(interpreterManager, me, none) .
rewrites: 5
result Configuration: <> < me : User | none > parsedQidList(me, interpreter(0),
    {'_-_['_/_['s_['0.Zero],'s_^2['0.Zero]],'s_['0.Zero]],'Rat})
==========================================
erewrite in PARSE-TEST : <> < me : User | qids: ('1/2 '- '1 '- '1 '/ '2) >
    createInterpreter(interpreterManager, me, none) .
rewrites: 5
result Configuration: <> < me : User | none > parsedQidList(me, interpreter(0),
    {'_-_['_-_['_/_['s_['0.Zero],'s_^2['0.Zero]],'s_['0.Zero]],'_/_['s_[
    '0.Zero],'s_^2['0.Zero]]],'Rat})
==========================================
erewrite in PARSE-TEST : <> < me : User | qids: '0 > createInterpreter(
    interpreterManager, me, none) .
rewrites: 5
result Configuration: <> < me : User | none > parsedQidList(me, interpreter(0),
    ambiguity({'0.Zero,'Zero}, {'0.Foo,'Foo}))
Bye.
